src/mainwindow.cpp

```diff
@@ -3633,14 +3611,10 @@ void MainWindow::setUseNativeWindowFrame(bool useNativeWindowFrame)
         flags &= ~Qt::FramelessWindowHint;
     else
         flags |= Qt::FramelessWindowHint;
-#  elif _WIN32
-    if (useNativeWindowFrame) {
-        flags &= ~Qt::CustomizeWindowHint;
-        flags &= ~Qt::FramelessWindowHint;
-    } else {
+#  elif defined(Q_OS_WINDOWS)
+    flags = Qt::Window;
+    if (!useNativeWindowFrame)
         flags |= Qt::CustomizeWindowHint;
-        flags |= Qt::FramelessWindowHint;
```

Comment:
Hmm, this basically takes it back to the initial implementation I did back in e013bb14d873e87cbdd9278a4ff18f54300f50f0. For some reason `FramelessWindowHint` flag was added in 15fa87080fc4f450f512b78983f4a21a6579c66c, but I have no idea what "window ui problem" that was meant to address and whether this flag was needed for something (that flag was already removed on Windows a long time ago in 5831d787d385339cf0fe16104211ed894bb260dc).

Patch looks fine, but I would prefer to use a similar method as for the Linux case to set/unset the flag, to keep the possibility open that we or Qt might use other unrelated flags.

Also, you may want to adjust `MainWindow::setupMainWindow` to also include `Qt::Window` flag for consistency (or use `setWindowFlags(windowFlags() | Qt::CustomizeWindowHint)` instead.

Btw, the `QProgressDialog` in `MainWindow::InitData` uses `Qt::FramelessWindowHint`. Should it also be changed to `Qt::CustomizeWindowHint`?

Reply:
>Patch looks fine, but I would prefer to use a similar method as for the Linux case to set/unset the flag, to keep the possibility open that we or Qt might use other unrelated flags.

Will do!

> Also, you may want to adjust `MainWindow::setupMainWindow` to also include `Qt::Window` flag for consistency (or use `setWindowFlags(windowFlags() | Qt::CustomizeWindowHint)` instead.

Right, good catch. 

> Btw, the `QProgressDialog` in `MainWindow::InitData` uses `Qt::FramelessWindowHint`. Should it also be changed to `Qt::CustomizeWindowHint`?

I think so, yes. At least for consistency. And the updater window as well...